scripts/odb_extractor_atoms.py

```diff
@@ -154,104 +538,174 @@ def find_atoms(observation):
-                                                                                                       filter,
-                                                                                                       disperser,
-                                                                                                       wavelength, p, q,
-                                                                                                       atomlabel))
-    # print(atoms)
+        print('{:17} {:3} {:7.2f} {:3d} {:10} {:12} {:10} {:12} {:6.4f} {:8.2f} {:8.2f} {:1} {:3d}'.format(
+            shortid(datalab),
+            observe_class[0:3], exptimes[ii], coadds[ii], config['inst'], config['fpu'][ii],
+            config['filter'][ii], config['disperser'][ii], config['wavelength'][ii], poffsets[ii],
+            qoffsets[ii], guiding[-1], atomlabel),
+              file=fid)
+        if ws is not None:
+            # Columns
+            #             columns = ['datalab', 'class', 'type', inst', 'exec_time', 'exptime', 'coadds', 'fpu', 'filter',
+            #                        'disperser', 'wavelength', 'p', 'q', 'guiding', 'qa_state', 'atom']
+            data = [datalab, observe_class.upper(), step['observe:observeType'].upper(), config['inst'],
+                    step_time, exptimes[ii], coadds[ii], config['fpu'][ii],
+                    config['filter'][ii], config['disperser'][ii], config['wavelength'][ii], poffsets[ii],
+                    qoffsets[ii], guiding[-1], qastate.upper(), atomlabel]
+            for jj in range(len(columns)):
+                _ = ws.cell(column=jj + 1, row=row, value=data[jj])
+            row += 1
+
+    #     print(atoms)
     # Get class/state for last atom
     if natom > 0:
-        print(qastates)
-        atoms[-1]['qastate'] = select_qastate(qastates)
-        print(classes)
+        atoms[-1]['qa_state'] = select_qastate(qastates)
+        if atoms[-1]['qa_state'] != 'NONE':
+            atoms[-1]['observed'] = True
         atoms[-1]['class'] = select_obsclass(classes)
+        if verbose:
+            print('QA states: ', qastates, file=fid)
+            print('Classes: ', classes, file=fid)
+            print('Guiding: ', guiding, file=fid)
+        atoms[-1]['guide_state'] = any(guiding)
+        atoms[-1]['wavelength'] = config['wavelength'][ii]
+        atoms[-1]['required_resources']['inst'] = config['inst']
+        atoms[-1]['required_resources']['filter'] = config['filter'][ii]
+        atoms[-1]['required_resources']['disperser'] = config['disperser'][ii]
+        atoms[-1]['required_resources']['fpu'] = config['fpu'][ii]
+
+        # Print basic atom info
+        print(" \t exec_time: {:7.2f}, prog_time: {:7.2f}, part_time: {:7.2f}, guide_state: {}".
+              format(atoms[-1]['exec_time'], atoms[-1]['prog_time'], atoms[-1]['part_time'],
+                     atoms[-1]['guide_state']), file=fid)
```

Comment:
The new Python standard is to eliminate `.format` and instead use string interpolation:

```python
print(f' \t exec_time: {atoms[-1]['exec_time']:7.2f} .....')
```
and also maybe store `atoms[-1]` in a variable to make this whole bit easier and then push it in?